packages/web-app-files/tests/unit/views/SharedViaLink.spec.js

```diff
@@ -1,29 +1,201 @@
+import { shallowMount, mount } from '@vue/test-utils'
+import flushPromises from 'flush-promises'
+import { getStore, localVue, createFile } from './views.setup.js'
+import SharedViaLink from '@files/src/views/SharedViaLink.vue'
+
+const $route = {
+  params: {
+    fileId: '2147491323'
+  },
+  meta: {
+    title: 'Resolving private link'
+  }
+}
+
+const activeFiles = [createFile({ id: 1234, type: 'file' })]
+
+localVue.prototype.$client.requests = {
+  ocs: jest.fn(() =>
+    Promise.resolve({
+      status: 200,
+      json: jest.fn(() =>
+        Promise.resolve({
+          ocs: {
+            data: activeFiles
+          }
+        })
+      )
+    })
+  )
+}
+
+const stubs = {
+  'router-link': true,
+  'no-content-message': true,
+  pagination: true,
+  'oc-table-files': true,
+  'context-actions': true,
+  'list-info': true
+}
+
+const selectors = {
+  listLoader: 'list-loader-stub',
+  noContentMessage: '#files-shared-via-link-empty',
+  ocTableFiles: '#files-shared-via-link-table',
+  contextActions: 'context-actions-stub',
+  listInfo: 'list-info-stub'
+}
+
 describe('SharedViaLink view', () => {
+  jest
+    .spyOn(SharedViaLink.mixins[3].methods, '$_filesListPagination_updateCurrentPage')
```

Comment:
if the order of the mixin get changed, this can fail. what about directly mocking from the mixin like:
```suggestion
    .spyOn(someMixin.methods, '$_filesListPagination_updateCurrentPage')
```